### DISCUSSION TITLE
USGS Topo tiles

### DISCUSSION THREAD

**Discussion by gregorypenn** [ropensci/terrainr#62] [Ideas]
I've been looking for an easy way to add USGS Topo tiles to maps made using ggplot2. Since the National Map server provides USGS topos, and terrainr gets map tiles from the National Map server, it seems like terrainr would be a natural way for me to add a USGS topo layer to my ggplots. Is there any reason this would be a difficult feature to implement?

**Comment by mikemahoney218** [comment]
I'd agree that it's a natural way to do this :smile: 

Have you checked out this vignette's section on using elevation data from terrainr with ggplot2?
https://docs.ropensci.org/terrainr/articles/overview.html

If there's anything not clear on how to apply this to your own situation, let me know -- but you should be able to use this data with ggplot2.

**Reply by gregorypenn** [reply]
That's a great vignette, which I had already looked over. Thanks for providing it! I'm looking for map tiles that provide more annotation and context than just topographic contours, though. That's why I'd prefer to get the USGS Topo map tiles rather than just creating my own topo contours from elevation data.

**Reply by mikemahoney218** [reply]
_Ah_, got it -- I misunderstood what you meant by topo tiles (thought you just meant the elevation data).

Let me look into adding the Topo tile endpoint and get back to you!

**Comment by mikemahoney218** [comment]
Hey @gregorypenn -- I added the following endpoints to the GitHub version of the package:

USGSTopo, USGSShadedReliefOnly, USGSImageryOnly, USGSHydroCached, USGSTNMBlank

The USGS Topo layer definitely works. The other ones sometimes work. I'm trying to see how much of that is a terrainr thing, and how much of that is the source APIs being a bit flaky as of late. Thanks for the suggestion!

**Reply by gregorypenn** [reply]
Thank you!!!

**Reply by mikemahoney218** [reply]
Thanks for the suggestion! This is now in the CRAN version (0.7.3)
